scripts/legacy_expression_tool.py

```diff
@@ -0,0 +1,394 @@
+# 
+#  Copyright 2023 Telefonica Investigación y Desarrollo, S.A.U
+# 
+#  This file is part of fiware-iotagent-lib
+# 
+#  fiware-iotagent-lib is free software: you can redistribute it and/or
+#  modify it under the terms of the GNU Affero General Public License as
+#  published by the Free Software Foundation, either version 3 of the License,
+#  or (at your option) any later version.
+# 
+#  fiware-iotagent-lib is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero General Public License for more details.
+# 
+#  You should have received a copy of the GNU Affero General Public
+#  License along with fiware-iotagent-lib.
+#  If not, see http://www.gnu.org/licenses/.
+# 
+#  Author by: Miguel Angel Pedraza
+# 
+
+from pymongo import MongoClient
+from bson import json_util, ObjectId
+import json
+from operator import itemgetter
+import re
+
+from datetime import datetime
+
+import argparse
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def parse_json(data):
+    return json.loads(json_util.dumps(data))
+
+object_replaced_list = []
+find_object_occurrences = []
+found_legacy_expressions = []
+translation_legacy_expressions = []
+object_occurrences_backup = []
+
+# Init time
+now = datetime.now()
+init_time = now.strftime("%y-%m-%d %H:%M:%S")
+
+# Create the CLI argunments parser
+parser = argparse.ArgumentParser(description='Tool to migrate legacy expressions in IoT Agents')
+parser.add_argument('-db','--database', help='Database name', required=True)
+parser.add_argument('-c','--collection', help='Collection name', required=True)
+parser.add_argument('-H','--host', help='Mongodb host', required=False)
+parser.add_argument('-p','--port', help='Mongodb port', required=False)
+parser.add_argument('-t','--translation', help='Translation file', required=False)
+parser.add_argument('-D','--debug', help='Debug mode', required=False, action='store_true')
+parser.add_argument('-cm','--commit', help='Commit changes to database', required=False, action='store_true')
+parser.add_argument('-el','--expressionlanguage', help='How to handle expressionLanguage values. Can be: delete, ignore or jexl', required=False)
+parser.add_argument('-stat','--statistics', help='Show statistics at the end of the execution. Possible values: service subservice', required=False)
+parser.add_argument('-S','--service', help='FIWARE service filter', required=False)
+parser.add_argument('-P','--servicepath', help='FIWARE servicepath filter', required=False)
+parser.add_argument('-i','--deviceid', help='Device ID filter', required=False)
+parser.add_argument('-r','--restore', help='Restore from backup file', required=False)
```

Comment:
In line with my previous suggestion:

```suggestion
parser.add_argument('--database', help='Database name', required=True)
parser.add_argument('--collection', help='Collection name', required=True)
parser.add_argument('--host', help='Mongodb host', required=False)
parser.add_argument('--port', help='Mongodb port', required=False)
parser.add_argument('--translation', help='Translation file', required=False)
parser.add_argument('--debug', help='Debug mode', required=False, action='store_true')
parser.add_argument('--commit', help='Commit changes to database', required=False, action='store_true')
parser.add_argument('--expressionlanguage', help='How to handle expressionLanguage values. Can be: delete, ignore or jexl', required=False)
parser.add_argument('--statistics', help='Show statistics at the end of the execution. Possible values: service subservice', required=False)
parser.add_argument('--service', help='FIWARE service filter', required=False)
parser.add_argument('--servicepath', help='FIWARE servicepath filter', required=False)
parser.add_argument('--deviceid', help='Device ID filter', required=False)
parser.add_argument('--restore', help='Restore from backup file', required=False)
```

Reply:
Fixed in https://github.com/telefonicaid/iotagent-node-lib/pull/1381/commits/e64aa3d0f747d3af9226bc2e70d66540c82b8115